test/framework/machinehealthcheck_helpers.go

```diff
@@ -0,0 +1,194 @@
+			Expect(err).ToNot(HaveOccurred())
+
+			ms := &clusterv1.MachineSetList{}
+			Expect(input.ClusterProxy.GetClient().List(ctx, ms, client.InNamespace(input.Cluster.Namespace), client.MatchingLabels(selectorMap))).To(Succeed())
+
+			if len(ms.Items) == 0 {
+				continue
+			}
+			machineSet := ms.Items[0]
+			selectorMap, err = metav1.LabelSelectorAsMap(&machineSet.Spec.Selector)
+			Expect(err).ToNot(HaveOccurred())
+
+			machines := &clusterv1.MachineList{}
+			Expect(input.ClusterProxy.GetClient().List(ctx, machines, client.InNamespace(machineSet.Namespace), client.MatchingLabels(selectorMap))).To(Succeed())
+
+			for _, machine := range machines.Items {
+				machineCopy := machine.DeepCopy()
+				patchHelper, err := patch.NewHelper(machineCopy, input.ClusterProxy.GetClient())
+				Expect(err).ToNot(HaveOccurred())
+				machine.SetLabels(mhc.Spec.Selector.MatchLabels)
+				Expect(patchHelper.Patch(ctx, machineCopy)).To(Succeed())
+			}
+		}
+
+		fmt.Fprintf(GinkgoWriter, "Ensuring there is at least 1 Machine that MachineHealthCheck is matching\n")
+		machines := GetMachinesByMachineHealthCheck(context.TODO(), GetMachinesByMachineHealthCheckInput{
+			Lister:             input.ClusterProxy.GetClient(),
+			ClusterName:        input.Cluster.Name,
+			Namespace:          input.Cluster.Namespace,
+			MachineHealthCheck: mhc,
+		})
+		if len(machines) == 0 {
+			continue
+		}
+
+		fmt.Fprintf(GinkgoWriter, "Patching MachineHealthCheck unhealthy condition to one of the nodes\n")
+		unhealthyNodeCondition := corev1.NodeCondition{
+			Type:               mhc.Spec.UnhealthyConditions[0].Type,
+			Status:             mhc.Spec.UnhealthyConditions[0].Status,
+			LastTransitionTime: metav1.Time{Time: time.Now()},
+		}
+		PatchNodeCondition(ctx, PatchNodeConditionInput{
+			ClusterProxy:  input.ClusterProxy,
+			Cluster:       input.Cluster,
+			NodeCondition: unhealthyNodeCondition,
+			Machine:       machines[0],
+		})
+	}
+
+	fmt.Fprintf(GinkgoWriter, "Waiting for remediation\n")
+	WaitForMachineHealthCheckToRemediateUnhealthyNodeCondition(ctx, WaitForMachineHealthCheckToRemediateUnhealthyNodeConditionInput{
+		ClusterProxy:        input.ClusterProxy,
+		Cluster:             input.Cluster,
+		MachineHealthChecks: machineHealthChecks,
+	}, input.WaitForMachineRemediation...)
+}
+
+// GetMachineHealthChecksByClusterInput is the input for GetMachineHealthChecksByCluster.
+type GetMachineHealthChecksByClusterInput struct {
+	Lister      Lister
+	ClusterName string
+	Namespace   string
+}
+
+// GetMachineHealthChecksByCluster returns the MachineHealthCheck objects for a cluster.
+// Important! this method relies on labels that are created by the CAPI controllers during the first reconciliation, so
+// it is necessary to ensure this is already happened before calling it.
+func GetMachineHealthChecksByCluster(ctx context.Context, input GetMachineHealthChecksByClusterInput) []*clusterv1.MachineHealthCheck {
```

Comment:
```suggestion
func GetMachineHealthChecksForCluster(ctx context.Context, input GetMachineHealthChecksByClusterInput) []*clusterv1.MachineHealthCheck {
```

`ByCluster` would be the name you'd use if it returns `map[Cluster][]MachineHealthCheck`